### DISCUSSION TITLE
Dual- or multi-language mode

### DISCUSSION THREAD

**Discussion by w-sky** [AnySoftKeyboard/AnySoftKeyboard#3644] [Ideas]
Hi there, I am constantly switching between two languages and with my previous keyboard I was used to a multi-language mode that would show me the keyboard of the non-english language (which has more letters), but auto-suggest dictionary and gesture typing worked for this language and English.

So this is my suggestion. I know there are ways to quickly change the keyboard language, but a dual-language (or multi-language) mode would be even more comfortable.

**Comment by fromSmolsoft** [comment]
+1

**Comment by woj-tek** [comment]
There are a couple of issues about it:

https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/908
https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/832
https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1094
https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/909
https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1590
https://github.com/AnySoftKeyboard/AnySoftKeyboard/issues/1727

Though most of them are closed... There is a hack to enable multiple languages at the same time but it's cumbersome and doesn't work all that well..


For me it's the single feature that holds me back from switching from SwiftKey (though, SK is getting exponentially terrible…)...

**Reply by menny** [reply]
Currently, to get this behavior you will need to manually override the default dictionary for the active keyboard layout.

I can think of one enhancement to this, by having a setting that tells AnySoftKeyboard to ALWAYS use ALL available dictionaries regardless of which layout is enabled. Does this sound like something desirable?

**Reply by woj-tek** [reply]
All *installed* dictionaries? IMHO it would work in my case (also - having single layout with additional custom characters from installed dictionaries would make it perfect but that's just cherry on top :-) )
